src/encoded/types/quality_metric.py

```diff
@@ -313,6 +313,49 @@ def tooltip(numVal):
         return qc_summary
 
 
+@collection(
+    name='quality-metrics-chipseq-v2',
+    properties={
+        'title': 'QC Quality Metrics for ChIP-seq',
+        'description': 'Listing of QC Quality Metrics for ChIP-seq',
+    })
+class QualityMetricChipseqV2(QualityMetric):
+    """Subclass of quality metrics for chip-seq"""
+
+    item_type = 'quality_metric_chipseq_v2'
+    schema = load_schema('encoded:schemas/quality_metric_chipseq_v2.json')
+    embedded_list = QualityMetric.embedded_list
+
+    @calculated_property(schema=QC_SUMMARY_SCHEMA)
+    def quality_metric_summary(self, request):
+        qc = self.properties
+        qc_summary = []
+
+        if 'lib_complexity' in qc:
+
+            qc_summary.append({"title": "PCR Bottleneck Coefficient (PBC)",
+                               "value": str(round2(qc.get("lib_complexity")["lib_complexity"]["rep1"]["PBC1"])),
+                               "tooltip": "one-read non-mito read pairs / distinct non-mito read pairs",
+                               "numberType": "float"})
+            
+            qc_summary.append({"title": "Nonredundant Read Fraction (NRF)",
+                               "value": str(round2(qc.get("lib_complexity")["lib_complexity"]["rep1"]["NRF"])),
+                               "tooltip": "distinct non-mito read pairs / total non-mito read pairs",
+                               "numberType": "float"})
+
+        if 'align' in qc:
+            
+            final_reads = qc.get(
+                "align")["nodup_samstat"]["rep1"]["read1"]  # PE
```

Comment:
If align is not present in qc, then `qc.get("align")` returns None and the following superscript ["nodup_samstat"] etc makes no sense and returns an error.
Either you are sure that "align" is always found (and you can just do `qc["align"]["nodup_samstat"]`), or you need to make it so it does not throw an error.

I see that you used the same style in other places in this calcprop, and I think all of those should be fixed.

Reply:
Oh, sorry, I did not see the line above: `if 'align' in qc`. I guess I'm not sure why you do a get, but it should work.

Reply:
I'm still not sure, in case of SE, what happens at this line. Will there not be a "read1"? Or do you always have a "read1" key which can however have a null value?

Reply:
In both cases of SE and PE, `qc["align"]["nodup_samstat"]["rep1"]["read1"]` exists and has a value. In the case of SE, the value is zero. The QC metric attached to the processed file from the fastq 4DNFIWQXK28E is one example.